### PR TITLE
Support for dedicated per validator Redis cache

### PR DESCRIPTION
**Pull request by atrifan**
@cristianconstantin, @andralungu

### CODE REVIEW

File: src/lua/api-gateway/validation/oauth2/userProfileValidator.lua [hunk 1]
```diff
@@ -173,7 +175,6 @@ function _M:validateUserProfile()
             cachedUserProfile = cjson.decode(cachedUserProfile)
         end
         self:setContextProperties(self:getContextPropertiesObject(cachedUserProfile))
-        if ( self:isProfileValid(cachedUserProfile) == true ) then
```
Comment (cristianconstantin) [inline src/lua/api-gateway/validation/oauth2/userProfileValidator.lua]: Was this line accidentally removed?

File: src/lua/api-gateway/validation/validator.lua [hunk 2]
```diff
@@ -79,6 +80,9 @@ function BaseValidator:setKeyInLocalCache(key, string_value, exptime, dict_name)
 end
 
 function BaseValidator:getRedisUpstream(upstream_name)
+    if self.redisHost then
```
Comment (cristianconstantin) [inline src/lua/api-gateway/validation/validator.lua]: I believe you also need a second variable, for the Redis read replica.
I would rather change the static `redis_RO_upstream` and `redis_RW_upstream` variables to class members of the `"api-gateway.validation.validator"` base class have them overwritten when this class is extended by `oauthTokenValidator`.

File: src/lua/api-gateway/validation/oauth2/userProfileValidator.lua [hunk 3]
```diff
@@ -173,7 +176,6 @@ function _M:validateUserProfile()
...
```
Comment (cristianconstantin) [inline src/lua/api-gateway/validation/oauth2/userProfileValidator.lua]: Was this line accidentally removed?

**Review by cristianconstantin — CHANGES_REQUESTED** [review]

File: src/lua/api-gateway/validation/oauth2/userProfileValidator.lua (hunk 3 above)
Comment (atrifan) [reply src/lua/api-gateway/validation/oauth2/userProfileValidator.lua]: it's fixed now can u check again? Thanks

**Comment by ddragosd** [comment]
👍  LGTM. didn't test it though.

**Comment by atrifan** [comment]
@ddragosd existing tests are passing are you referring at adding additional tests to check upstreams?

File: dist/maven/pom.xml [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
     <groupId>com.adobe.api.gateway</groupId>
     <artifactId>api-gateway-request-validation</artifactId>
-    <version>1.2.7-SNAPSHOT</version>
+    <version>1.2.8-SNAPSHOT</version>
```
Comment (andralungu) [inline dist/maven/pom.xml:20]: I guess this gets automagically updated by the maven release plugin.

**Review by andralungu — APPROVED** [review]

**Review by cristianconstantin — APPROVED** [review]

**Pull request merged**